mars/services/task/supervisor/processor.py

```diff
@@ -39,15 +40,20 @@
 from .preprocessor import TaskPreprocessor
 from .stage import TaskStageProcessor
 
+logger = logging.getLogger(__name__)
 
-def _record_error(func: Union[Callable, Coroutine]):
+
+def _record_error(func: Union[Callable, Coroutine]=None, log_when_error=True):
     assert asyncio.iscoroutinefunction(func)
 
     @wraps(func)
     async def inner(processor: "TaskProcessor", *args, **kwargs):
         try:
             return await func(processor, *args, **kwargs)
         except:  # noqa: E722  # nosec  # pylint: disable=bare-except  # pragma: no cover
+            _, err, tb = sys.exc_info()
+            if log_when_error:
+                logger.exception('Error happens in %s', func, exc_info=sys.exc_info())
```

Comment:
`logger.exception` will include error stack automatically. Thus `exc_info` argument is not needed.